src/veileder/result/KandidatsøkFraStilling.tsx

```diff
@@ -59,44 +62,39 @@ const KandidatsøkFraStilling: FunctionComponent<Props> = ({
     search,
     harHentetStilling,
     maksAntallTreff,
-    kandidatlisteIdFraSøk,
     fjernValgtKandidat,
+    stillingsId,
 }) => {
+    const stillingsIdFraUrl = match.params.stillingsId;
+
     useEffect(() => {
         window.scrollTo(0, 0);
         resetKandidatlisterSokekriterier();
     }, [resetKandidatlisterSokekriterier]);
 
-    const stillingsId = match.params.stillingsId;
-
     useEffect(() => {
         fjernValgtKandidat();
     }, [fjernValgtKandidat]);
 
     useEffect(() => {
-        const søkestateKommerFraDenneKandidatlisten =
-            !!kandidatlisteIdFraSøk && kandidatlisteIdFraSøk === kandidatliste?.kandidatlisteId;
-
-        const skalSøkeMedEksisterendeSøkestate =
-            !harUrlParametere(window.location.href) && søkestateKommerFraDenneKandidatlisten;
-
-        if (skalSøkeMedEksisterendeSøkestate) {
-            search();
-        } else if (harUrlParametere(window.location.href)) {
+        if (harUrlParametere(window.location.href)) {
             leggUrlParametereIStateOgSøk(window.location.href, kandidatliste?.kandidatlisteId);
-        } else if (!harHentetStilling) {
-            leggInfoFraStillingIStateOgSøk(stillingsId, kandidatliste?.kandidatlisteId);
+        } else {
+            leggInfoFraStillingIStateOgSøk(stillingsIdFraUrl, kandidatliste?.kandidatlisteId);
         }
-        // eslint-disable-next-line
     }, [
         kandidatliste,
-        kandidatlisteIdFraSøk,
-        stillingsId,
-        harHentetStilling,
+        stillingsIdFraUrl,
         leggInfoFraStillingIStateOgSøk,
         leggUrlParametereIStateOgSøk,
     ]);
 
+    useEffect(() => {
+        if (!harUrlParametere(window.location.href)) {
+            hentKandidatlisteMedStillingsId(stillingsIdFraUrl);
```

Comment:
Kan denne flyttes inn i en andre useEffecten, det blir vel samme trigger for dem begge? Når stilling skal hentes, skal også kandidatlisten hentes

Reply:
Om de er sammen har vi ikke noe exit criteria, blir mange kall mot backend, før den gir opp. Kandidatlisten vil da oppdateres og trigges om og om igjen.